src/openapi/jolokia/queries/index.ts

```diff
@@ -0,0 +1,489 @@
+    jolokiaTargetPodSessionId: number;
+  },
+  queryKey?: TQueryKey,
+  options?: Omit<
+    UseQueryOptions<
+      Awaited<ReturnType<typeof JolokiaService.getBrokerDetails>>,
+      unknown,
+      Awaited<ReturnType<typeof JolokiaService.getBrokerDetails>>,
+      unknown[]
+    >,
+    'queryKey' | 'queryFn' | 'initialData'
+  >,
+) =>
+  useQuery(
+    [
+      useJolokiaServiceGetBrokerDetailsKey,
+      ...(queryKey ?? [{ jolokiaSessionId, jolokiaTargetPodSessionId }]),
+    ],
+    () =>
+      JolokiaService.getBrokerDetails(
+        jolokiaSessionId,
+        jolokiaTargetPodSessionId,
+      ),
+    options,
+  );
+export const useJolokiaServiceReadBrokerAttributesKey =
+  'JolokiaServiceReadBrokerAttributes';
+export const useJolokiaServiceReadBrokerAttributes = <
+  TQueryKey extends Array<unknown> = unknown[],
+>(
+  {
+    jolokiaSessionId,
+    jolokiaTargetPodSessionId,
+    names,
+  }: {
+    jolokiaSessionId: string;
+    jolokiaTargetPodSessionId: number;
+    names?: Array<string>;
+  },
+  queryKey?: TQueryKey,
+  options?: Omit<
+    UseQueryOptions<
+      Awaited<ReturnType<typeof JolokiaService.readBrokerAttributes>>,
+      unknown,
+      Awaited<ReturnType<typeof JolokiaService.readBrokerAttributes>>,
+      unknown[]
+    >,
+    'queryKey' | 'queryFn' | 'initialData'
+  >,
+) =>
+  useQuery(
+    [
+      useJolokiaServiceReadBrokerAttributesKey,
+      ...(queryKey ?? [{ jolokiaSessionId, jolokiaTargetPodSessionId, names }]),
+    ],
+    () =>
+      JolokiaService.readBrokerAttributes(
+        jolokiaSessionId,
+        jolokiaTargetPodSessionId,
+        names,
+      ),
+    options,
+  );
+export const useJolokiaServiceCheckCredentialsKey =
+  'JolokiaServiceCheckCredentials';
+export const useJolokiaServiceCheckCredentials = <
+  TQueryKey extends Array<unknown> = unknown[],
+>(
+  {
+    jolokiaSessionId,
+    jolokiaTargetPodSessionId,
+  }: {
+    jolokiaSessionId: string;
+    jolokiaTargetPodSessionId: number;
+  },
+  queryKey?: TQueryKey,
+  options?: Omit<
+    UseQueryOptions<
+      Awaited<ReturnType<typeof JolokiaService.checkCredentials>>,
+      unknown,
+      Awaited<ReturnType<typeof JolokiaService.checkCredentials>>,
+      unknown[]
+    >,
+    'queryKey' | 'queryFn' | 'initialData'
+  >,
+) =>
+  useQuery(
+    [
+      useJolokiaServiceCheckCredentialsKey,
+      ...(queryKey ?? [{ jolokiaSessionId, jolokiaTargetPodSessionId }]),
+    ],
+    () =>
+      JolokiaService.checkCredentials(
+        jolokiaSessionId,
+        jolokiaTargetPodSessionId,
+      ),
+    options,
+  );
+export const useJolokiaServiceExecBrokerOperation = (
+  options?: Omit<
+    UseMutationOptions<
+      Awaited<ReturnType<typeof JolokiaService.execBrokerOperation>>,
+      unknown,
+      {
+        jolokiaSessionId: string;
+        jolokiaTargetPodSessionId: number;
+        requestBody: OperationRef;
+      },
+      unknown
+    >,
+    'mutationFn'
+  >,
+) =>
+  useMutation(
```

Comment:
@lavocatt , what's the useMutation hook does here?

Reply:
There is two different kind of hooks provided by `react-query`. The mutations and the queries. Mutations are used to perform an operation that might cause changes on the server, queries are used to gather information from the server. You can see queries as something that needs to be refetched periodically, vs a mutation is something to execute only when needed.

In the PR, a mutation is used for login as it's performing a server side operation and a query is used to get the api-info since this operation is only gathering data from the server to the client.

Maybe some better explanation over there:
https://tanstack.com/query/v4/docs/framework/react/guides/queries
https://tanstack.com/query/v4/docs/framework/react/guides/mutations

Reply:
Okay, got it. Thanks for the explanation.